packages/wrangler/src/__tests__/whoami.test.tsx

```diff
@@ -226,5 +227,63 @@ describe("WhoAmI component", () => {
 		expect(lastFrame()).toMatch(/Account One .+ account-1/);
 		expect(lastFrame()).toMatch(/Account Two .+ account-2/);
 		expect(lastFrame()).toMatch(/Account Three .+ account-3/);
+		expect(lastFrame()).toContain(
+			"Token Permissions: If scopes are missing, you may need to logout and re-login."
+		);
+		expect(lastFrame()).toContain("- scope1 (read)");
+		expect(lastFrame()).toContain("- scope2 (write)");
+		expect(lastFrame()).toContain("- scope3");
+	});
+
+	// For the case where the cache hasn't updated to include the scopes array
+	it("should display the user's email and accounts, but no OAuth scopes if none provided", async () => {
+		const user: UserInfo = {
+			authType: "OAuth Token",
+			apiToken: "some-oauth-token",
+			email: "user@example.com",
+			tokenPermissions: undefined,
+			accounts: [
+				{ name: "Account One", id: "account-1" },
+				{ name: "Account Two", id: "account-2" },
+				{ name: "Account Three", id: "account-3" },
+			],
+		};
+
+		const { lastFrame } = render(<WhoAmI user={user}></WhoAmI>);
+
+		expect(lastFrame()).toContain(
+			"You are logged in with an OAuth Token, associated with the email 'user@example.com'!"
+		);
+		expect(lastFrame()).toMatch(/Account Name .+ Account ID/);
+		expect(lastFrame()).toMatch(/Account One .+ account-1/);
+		expect(lastFrame()).toMatch(/Account Two .+ account-2/);
+		expect(lastFrame()).toMatch(/Account Three .+ account-3/);
```

Comment:
Any reason why you didn't use `expect(lastFrame()).toMatchInlineSnapshot(...)` here and below?

Reply:
This was what was being used previously here, I assumed there was a reason that they had decided to write it this way

Reply:
It may have been that there was only a certain thing the person was looking for the output to match. I think Pete is implying it would make more sense to check all those values at once, both ways technically work.

Reply:
Looks good! Merge away 😄